libs/libc/machine/sim/arch_setjmp_arm64.S

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ * libs/libc/machine/sim/arch_setjmp_arm64.S
+ *
+ *   Copyright (C) 2017 Gregory Nutt. All rights reserved.
+ *   Author: Nickolay Semyonov (RPI) <snob@wolpike.com>
+ *
+ * Extracted from the MUSL C-library.  The MUSL C library has a compatible
+ * MIT license and is released here under the NuttX 3-clause BSD license:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_HOST_MACOS)
+	#define SYMBOL(x) _##x
+#else
+	#define SYMBOL(x) x
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+	.text
+	.globl SYMBOL(setjmp)
+	.align 4
+
+SYMBOL(setjmp):
+
+	mov x1,x0						/* [0:]: */
+	mov x2,fp						/* [1:JB_FP]: FP (frame pointer) */
+	stp x1,x2,[x0]
+
+	mov x1,sp						/* [2:JB_SP]: SP (stack pointer) */
```

Comment:
does the code save enough register?

Reply:
It's currently saving, FP, SP and LR. I agree, that it sounds not enough ... but it's initially working.

Reply:
Can vfork test in ostest run without issue?

Reply:
Not yet done ... is ostest overall a good validation, that simulation is fully working?

Reply:
Yes, x86/x84 version pass all validation.

Reply:
With current implementation vfork/ostest is not working. It’s not crashing, but it‘s also not passing. Further investigation is needed. I will update PR as it‘s done.

Reply:
With latest update octets/vfork() is working:

  End of test memory usage:
  VARIABLE  BEFORE   AFTER
  ======== ======== ========
  arena     3fffe00  3fffe00
  ordblks         6        9
  mxordblk  3e02610  3db2cd0
  uordblks   1ebe10   24ccd0
  fordblks  3e13ff0  3db3130

  user_main: vfork() test
  vfork_test: Child 125 ran successfully

  Final memory usage:
  VARIABLE  BEFORE   AFTER
  ======== ======== ========
  arena     3fffe00  3fffe00
  ordblks         2        7
  mxordblk  3e389f0  3e04650
  uordblks   1c73d0   1e9c10
  fordblks  3e38a30  3e161f0
  user_main: Exiting
  ostest_main: Exiting with status 0

Reply:
Nice

Reply:
Let's merge the two patch into one.

Reply:
Definitely … I thought, I did it … I‘ll do it soon.